src/main/java/com/sprint/mission/discodeit/JFCTest.java

```diff
@@ -0,0 +1,223 @@
+import com.sprint.mission.discodeit.service.jcf.JFCChannelService;
+import com.sprint.mission.discodeit.service.jcf.JFCMessageService;
+import com.sprint.mission.discodeit.service.jcf.JFCUserService;
+
+import java.util.List;
+
+public class JFCTest {
+    private static final JFCUserService userService = JFCUserService.getInstance();
+    private static final JFCMessageService messageService = JFCMessageService.getInstance();
+    private static final JFCChannelService channelService = JFCChannelService.getInstance();
+
+    private static final UserChannelService userChannelService = UserChannelService.getInstance();
+
+    public static void main(String[] args) {
+
+
+        User user1 = userService.createUser("짱구", "자는 중", "aaa@gmail.com");
+        User user2 = userService.createUser("맹구", "식사 중", "bbb@gmail.com");
+        User user3 = userService.createUser("유리", "공부 중", "ccc@gmial.com");
+        User user4 = userService.createUser("철수", "쉬는 중", "ddd@gmial.com");
+        User user5 = userService.createUser("훈이", "씻는 중", "eee@gmial.com");
+
+        Channel channel1 = channelService.createChannel("떡잎 마을 방범대", "떡잎 마을을 지키는 모임");
+        Channel channel2 = channelService.createChannel("붉은 장미 삼총사", "정의로운 모임");
+
+        printAllChannels(); // 모든 채널 출력
+        printAllUsers(); // 모든 유저 출력
+
+        joinChannel(user1, channel1);
+        joinChannel(user2, channel1);
+        joinChannel(user2, channel2); // user2(맹구)의 채널 중복 입장
+        joinChannel(user4, channel2);
+
+        printUsersInChannel(channel1); // channel1(떡잎 마을 방범대) 유저 출력
+        printChannelsOfUser(user2); // user2(맹구)가 속한 채널 출력
+
+        sendMessage(user1, channel1, "출동!");
+        sendMessage(user2, channel1, "발사!");
+        sendMessage(user4, channel1, "출동!"); // 속하지 않은 채널에 메시지 전송
+
+        printMessagesInChannel(channel1);
+        printMessagesInChannel(channel2); // 메시지 0개인 채널 메시지 조회 시도
+
+        updateChannelInfo(channel1, "떡잎 마을 파괴!");
+        printChannel(channel1); // 채널 정보 수정 후 채널 출력
+
+        updateChannelName(channel1, "떡잎 마을 파괴단");
+        printMessagesInChannel(channel1); // 채널명 변경 후 메시지 조회
+
+        deleteUser(user1);
+        printMessagesInChannel(channel1); // user1(짱구) 삭제 후 메시지 출력
+
+        printAllUsers();
+
+        deleteChannel(channel1);
+        printMessagesOfUser(user2); // 채널 삭제 후 해당 채널에서 채팅한 유저의 메시지 출력
+
+    }
+
+
+    // -----------------------------------------------
+    // 채널 입퇴장
+    private static void joinChannel(User user, Channel channel){
+        if(userService.notExist(user) || channelService.notExist(channel)){
+            System.out.println("[Error] 존재하지 않는 유저 혹은 채널입니다.\n");
+        } else{
+            if(!userChannelService.put(user.getId(), channel.getId())){
+                System.out.println("[Error] 이미 입장한 채널입니다.\n");
+            }else {
+                System.out.println("[Info] " + user.getName() + " 님이 " + channel.getName()
+                        + "에 입장했습니다.\n");
+            }
+        }
+    }
```

Comment:
if-else 문을 중첩으로 사용해주셨네요 👍   

취향 차이일 수 있겠지만, 저는 이렇게 중간중간에 ealry return을 사용해서 들여쓰기를 줄이는 편이 가독성이 더 좋더라구요.  

```suggestion
    private static void joinChannel(User user, Channel channel) {
        if (userService.notExist(user) || channelService.notExist(channel)) {
            System.out.println("[Error] 존재하지 않는 유저 혹은 채널입니다.\n");
            return;
        }
        boolean isInvolved = userChannelService.put(user.getId(), channel.getId());
        if (!isInvolved) {
            System.out.println("[Error] 이미 입장한 채널입니다.\n");
            return;
        }
        
        System.out.println("[Info] " + user.getName() + " 님이 " + channel.getName() + "에 입장했습니다.\n");
    }
```